### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by m4dz**
This PR enables the logout feature in the settings menu

### CODE REVIEW

File: src/lib/stack.js [hunk 1]
```diff
@@ -5,52 +5,78 @@ import 'babel-polyfill'
...
+
+      try {
+        hasSettings = await hasApp('settings')
+      } catch (e) {
+        hasSettings = false
+        throw new UnavailableSettingsException()
+      }
+
+      if (!hasSettings) {
+        throw new UnavailableSettingsException()
```
Comment (CPatchane) [inline src/lib/stack.js]: Why throwing an error here? I think it should be considered as a normal behaviour that hasSettings is false, should it? It's expected as a function name `hasSomething` to respond `true` or `false` without error if `false` as it's a function to check an existence.

Comment (m4dz) [inline src/lib/stack.js]: I agree w/ your point, but as the Settings app is crucial (w/o it, no possibilities to update password nor change language) so I though throwing an Exception (as well as returning the status of the app, falsy) can be a pretty solution to interrupt Settings links rendering in view and display a wrning message in console about the disabling of those features. But I could be wrong…

**Comment by m4dz** [comment]
@CPatchane did you really review this PR when I pushed it at 01:00 am :fearful: ?

File: src/lib/stack.js (hunk 1 above)
Comment (CPatchane) [inline src/lib/stack.js]: Good point, I agree about this specific case of the settings app, maybe you can add a comment to be sure to not generalise this way if a similar function is necessary for another app? What do you think?

**Comment by CPatchane** [comment]
Great @m4dz thanks!

File: src/lib/stack.js (hunk 1 above)
Comment (m4dz) [inline src/lib/stack.js]: Comment added :taco:

**Pull request merged**